# WCToolbox-6.0.0-beta.1.zip pc/3do hash tables go in same folder
# as Wing Commander 3/4 tre files. CD files are lowercase.
diff a/wc3remake/PKGBUILD b/wc3remake/PKGBUILD
--- a/wc3remake/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
+++ b/wc3remake/PKGBUILD	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,66 @@
+# Maintainer: Noa Body <noabody@archlinux.org>
+
+pkgbase=wc3remake
+pkgname=("$pkgbase" wc4remake)
+pkgver=20261008.r0.g3d34e88
+pkgrel=1
+pkgdesc='Wing Commander libRealSpace.'
+arch=(x86_64)
+url=https://github.com/neuromancer/wc3remake
+license=(GPL-3.0-only)
+depends=(ffmpeg hicolor-icon-theme libglvnd perl-rename sdl2-compat sdl2_mixer_ext sdl2_ttf xorg-fonts-misc)
+makedepends=(cmake gendesk git gtest ninja openxr sdl2_mixer_ext sdl2_ttf stb simpleini)
+optdepends=('soundfont-fluid: FluidSynth soundfonts')
+source=(git+https://github.com/DMJC/"$pkgbase".git
+         https://raw.githubusercontent.com/noabody/unibuild/master/data/wc1-re.svg
+        "$pkgbase".patch
+        "$pkgbase".sh)
+sha256sums=('SKIP'
+            'b16d63e99a46f0196e8858e2ad77b3d1d703410332f94066d7a73a830d9136f5'
+            '45e1e78b54bc1b9496d7eb6f51b031990864a68ad77a956b1d96e09237f32dc2'
+            '37c240c375abc3dadcf94fb54ff037ba722538d1136251802efd182d33ef6563')
+
+pkgver() {
+  cd "$pkgbase"
+  printf "%s.r%s.g%s" "$(date +%Y%m%d)" "$(git rev-list "$(git rev-list --tags --no-walk --max-count=1)"..HEAD --count)" "$(git rev-parse --short=7 HEAD)"
+}
+
+prepare() {
+  perl -pe 's/wc3/wc4/gi' "$pkgbase".sh > wc4remake.sh
+  [[ ! -d "build" ]] || rm -rf build
+  sed -i -E 's/(target_link_libraries\([^)]+PRIVATE\s+)stb/\1/g' "$pkgbase"/src/executables/*/CMakeLists.txt
+  patch -Np1 -i "$pkgbase".patch
+}
+
+build() {
+  cmake -S "$pkgbase" -B build -G Ninja \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=/usr/share/"$pkgbase" \
+    -Wno-deprecated
+  cmake --build build -j$(nproc)
+}
+
+package_wc3remake() {
+  pkgdesc='Wing Commander 3 libRealSpace.'
+  gendesk -f -n --name='Wing Commander 3 lRS' --comment="${pkgdesc/.}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/$pkgname.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname",applications,icons/hicolor/scalable/apps}}
+  cp -dr --no-preserve='ownership' build/assets "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname"/resources/config.ini "$pkgdir"/usr/share/"$pkgname"/
+  install -m 755 build/src/executables/wc3/neowc3 "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 wc1-re.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname".svg
+  ln -rsf "$pkgdir"/usr/share/"$pkgname"/neowc3 "$pkgdir"/usr/bin/"$pkgname"
+}
+package_wc4remake() {
+  pkgdesc='Wing Commander 4 libRealSpace.'
+  gendesk -f -n --name='Wing Commander 4 lRS' --comment="${pkgdesc/.}" --custom='Keywords=game;engine;' --exec="/usr/share/$pkgname/$pkgname.sh" --icon="$pkgname" --terminal=false --categories='GNOME;GTK;Game' --startupnotify=false --pkgname="$pkgname"
+  install -dm 755 "$pkgdir"/usr/{bin,share/{"$pkgname",applications,icons/hicolor/scalable/apps}}
+  cp -dr --no-preserve='ownership' build/assets "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgbase"/resources/config.ini "$pkgdir"/usr/share/"$pkgname"/
+  install -m 755 build/src/executables/wc4/neowc4 "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 "$pkgname".desktop "$pkgdir"/usr/share/applications/
+  install -m 755 "$pkgname".sh "$pkgdir"/usr/share/"$pkgname"/
+  install -m 644 wc1-re.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname".svg
+  ln -rsf "$pkgdir"/usr/share/"$pkgname"/neowc4 "$pkgdir"/usr/bin/"$pkgname"
+}
diff a/wc3remake/wc3remake.sh b/wc3remake/wc3remake.sh
--- a/wc3remake/wc3remake.sh	1969-12-31 17:00:00.000000000 -0700
+++ b/wc3remake/wc3remake.sh	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,64 @@
+#!/bin/bash
+gmdir="$HOME/games/wc3"
+gmcfg="$HOME/.config/wc3remake"
+gmtgt="missions.tre"
+gmprm=""
+glnch="wc3remake"
+
+gconf() {
+mkdir -p "$gmcfg"
+printf 'gamepath=%s\ngamemod=%s\ngameparm=%s\n' "$gmdir" "$gmtgt" "$gmprm" > "$gmcfg/basedir"
+}
+
+gstart () {
+while IFS= read -r -d '' file; do
+  [[ -d "$gmbdr/$file" && -z "$gm1st" ]] || mkdir -p "$gmbdr/$file"
+done < <(find /usr/share/wc3remake -mindepth 1 -type d -printf '%P\0')
+while IFS= read -r -d '' file; do
+  [[ -f "$gmbdr/$file" && -z "$gm1st" ]] || cp -f "/usr/share/wc3remake/$file" "$gmbdr/$file"
+done < <(find /usr/share/wc3remake -mindepth 2 -type f ! \( -iname '*.ini' \) -printf '%P\0')
+[[ -f "$gmbdr/assets/config.ini" && -z "$gm1st" ]] || cp -f /usr/share/wc3remake/config.ini "$gmbdr/assets/config.ini"
+grep -Pioq "(?<=^data_path=)$gmbdr" "$gmbdr/assets/config.ini" || perl -pi -e "s#(?<=^data_path=).*#$gmbdr#gi" "$gmbdr/assets/config.ini"
+[[ -f "/usr/share/soundfonts/FluidR3_GM.sf2" ]] && export SDL_SOUNDFONTS="/usr/share/soundfonts/FluidR3_GM.sf2"
+echo "cd $gmbdr && $glnch $gmpar" | xargs -i -r sh -c "{}"
+}
+
+gtest () {
+gmbse="" gmbdr="" gmgme="" gmgnm="" gmgpm="" gmpar=""
+
+if [[ -f "$gmcfg/basedir" ]]; then
+  while IFS='=' read -r key val || [[ -n "$key" ]]; do
+    case "$key" in
+      gamepath) gmbse="gamepath=$val"; gmbdr="$val" ;;
+      gamemod)  gmgme="gamemod=$val";  gmgnm="$val" ;;
+      gameparm) gmgpm="gameparm=$val"; gmpar="$val" ;;
+    esac
+  done < "$gmcfg/basedir"
+fi
+}
+
+[[ -f "$gmcfg/basedir" ]] || { gconf; gm1st=1; }
+gtest
+if [[ -z "$gmbdr" ]]; then
+  gconf ; gm1st=1
+  gtest
+fi
+if [[ -d "$gmbdr" ]]; then
+  if [[ ! -f "$gmbdr/$gmgnm" ]]; then
+    gmchk="$(find -H "$gmbdr" -mindepth 1 -maxdepth 2 -type f -regextype posix-extended -iregex "$gmbdr/$gmgnm" -printf '%p\n' | grep -Pic "$gmbdr/$gmgnm")"
+    if (( $gmchk == 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt?\nRequires perl-rename command" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && find -H "$gmbdr" -depth -exec perl-rename 's|[^/]*\Z|\L$&|' {} +
+    elif (( $gmchk > 1 )) ; then
+      echo -e "File system is case sensitive\nFile/folder names must be lowercase\n Cannot rename subdirs of\n  $gmbdr\n    containing\n  $gmtgt\n\nBecause more than one exists" | xmessage -file - -buttons Ok:0 -center -timeout 30
+    else
+      echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n found, but not\n  $gmgme\n ($gmbdr/$gmgnm)\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+      (( $? == 101 )) && gconf
+    fi
+  else
+    gstart
+  fi
+else
+  echo -e "Config:\n $gmcfg/basedir\n  $gmbse\n not found\n\nRebuild config?" | xmessage -file - -buttons Yes,No -default No -center -timeout 30
+  (( $? == 101 )) && gconf
+fi
diff a/wc3remake/wc3remake.patch b/wc3remake/wc3remake.patch
index e69de29..feb733a 100644
--- a/wc3remake/wc3remake.patch	1969-12-31 17:00:00.000000000 -0700
+++ b/wc3remake/wc3remake.patch	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,141 @@
+diff a/wc3remake/CMakeLists.txt b/wc3remake/CMakeLists.txt
+--- a/wc3remake/CMakeLists.txt	1969-12-31 17:00:00.000000000 -0700
++++ b/wc3remake/CMakeLists.txt	1969-12-31 17:00:00.000000000 -0700
+@@ -4,35 +4,121 @@ set(CMAKE_CXX_STANDARD 17)
+ set(CMAKE_CXX_STANDARD_REQUIRED True)
+ add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+ 
+-# Trouver Google Test
+-find_package(GTest CONFIG REQUIRED)
++# Download and include CPM.cmake automatically
++set(CPM_DOWNLOAD_LOCATION "${CMAKE_BINARY_DIR}/cmake/CPM.cmake")
++if(NOT EXISTS "${CPM_DOWNLOAD_LOCATION}")
++    message(STATUS "Downloading CPM.cmake...")
++    file(DOWNLOAD
++        https://github.com/cpm-cmake/CPM.cmake/releases/latest/download/CPM.cmake
++        "${CPM_DOWNLOAD_LOCATION}"
++    )
++endif()
++include("${CPM_DOWNLOAD_LOCATION}")
+ 
+-# Activer les tests
++# System Dependencies
++find_package(GTest REQUIRED)
+ enable_testing()
+ 
+ find_package(SDL2 REQUIRED)
+-find_package(SDL2_mixer_ext CONFIG REQUIRED)
+ find_package(SDL2_ttf CONFIG REQUIRED)
+-find_package(imgui CONFIG REQUIRED)
+-# System imgui package puts backends in a subdirectory
+-if(EXISTS "/usr/include/imgui/backends")
+-    include_directories("/usr/include/imgui/backends")
+-endif()
+-# System imgui package exports IMGUI::IMGUI instead of vcpkg's imgui::imgui
+-if(NOT TARGET imgui::imgui AND TARGET IMGUI::IMGUI)
+-    add_library(imgui::imgui ALIAS IMGUI::IMGUI)
+-endif()
+ if (APPLE)
+     set(CMAKE_FIND_FRAMEWORK FIRST)
+ endif()
+ set (OpenGL_GL_PREFERENCE LEGACY)
+ find_package(OpenGL REQUIRED)
+ 
++# System SDL2_mixer_ext
++find_package(SDL2_mixer_ext QUIET)
++if(NOT TARGET SDL2_mixer_ext AND NOT TARGET SDL2_mixer_ext::SDL2_mixer_ext)
++    find_library(SDL2_MIXER_EXT_LIB NAMES SDL2_mixer_ext REQUIRED)
++    find_path(SDL2_MIXER_EXT_INCLUDE_DIR NAMES SDL_mixer_ext.h PATH_SUFFIXES SDL2 REQUIRED)
++    add_library(SDL2_mixer_ext UNKNOWN IMPORTED)
++    set_target_properties(SDL2_mixer_ext PROPERTIES
++        IMPORTED_LOCATION "${SDL2_MIXER_EXT_LIB}"
++        INTERFACE_INCLUDE_DIRECTORIES "${SDL2_MIXER_EXT_INCLUDE_DIR}"
++    )
++endif()
++
++if(TARGET SDL2_mixer_ext)
++    if(NOT TARGET SDL2_mixer_ext::SDL2_mixer_ext)
++        add_library(SDL2_mixer_ext::SDL2_mixer_ext ALIAS SDL2_mixer_ext)
++    endif()
++    if(NOT TARGET SDL2_mixer_ext::SDL2_mixer_ext_Static)
++        add_library(SDL2_mixer_ext::SDL2_mixer_ext_Static ALIAS SDL2_mixer_ext)
++    endif()
++endif()
++
++# In-tree ImGui & NFD via CPM
++CPMAddPackage(
++    NAME imgui
++    GITHUB_REPOSITORY ocornut/imgui
++    GIT_TAG v1.92.9
++    GIT_SHALLOW YES
++)
++if(NOT TARGET imgui::imgui)
++    add_library(imgui_static STATIC
++        "${imgui_SOURCE_DIR}/imgui.cpp"
++        "${imgui_SOURCE_DIR}/imgui_draw.cpp"
++        "${imgui_SOURCE_DIR}/imgui_tables.cpp"
++        "${imgui_SOURCE_DIR}/imgui_widgets.cpp"
++        "${imgui_SOURCE_DIR}/backends/imgui_impl_sdl2.cpp"
++        "${imgui_SOURCE_DIR}/backends/imgui_impl_opengl2.cpp"
++    )
++    target_include_directories(imgui_static PUBLIC "${imgui_SOURCE_DIR}" "${imgui_SOURCE_DIR}/backends")
++    target_link_libraries(imgui_static PUBLIC SDL2::SDL2)
++    add_library(imgui::imgui ALIAS imgui_static)
++endif()
++include_directories("${imgui_SOURCE_DIR}" "${imgui_SOURCE_DIR}/backends")
++
++CPMAddPackage(
++    NAME nfd
++    GITHUB_REPOSITORY btzy/nativefiledialog-extended
++    GIT_TAG v1.4.1
++    GIT_SHALLOW YES
++)
++if(TARGET nfd AND NOT TARGET nfd::nfd)
++    add_library(nfd::nfd ALIAS nfd)
++endif()
++# --- SDL2_mixer_ext ---
++find_package(SDL2_mixer_ext QUIET)
++if(NOT TARGET SDL2_mixer_ext AND NOT TARGET SDL2_mixer_ext::SDL2_mixer_ext)
++    find_library(SDL2_MIXER_EXT_LIB NAMES SDL2_mixer_ext)
++    find_path(SDL2_MIXER_EXT_INCLUDE_DIR NAMES SDL_mixer_ext.h PATH_SUFFIXES SDL2)
++    if(SDL2_MIXER_EXT_LIB AND SDL2_MIXER_EXT_INCLUDE_DIR)
++        add_library(SDL2_mixer_ext UNKNOWN IMPORTED)
++        set_target_properties(SDL2_mixer_ext PROPERTIES
++            IMPORTED_LOCATION "${SDL2_MIXER_EXT_LIB}"
++            INTERFACE_INCLUDE_DIRECTORIES "${SDL2_MIXER_EXT_INCLUDE_DIR}"
++        )
++    endif()
++endif()
++
++# --- Header-Only Libraries ---
++find_path(SIMPLEINI_INCLUDE_DIR NAMES SimpleIni.h PATHS /usr/include /usr/local/include)
++if(SIMPLEINI_INCLUDE_DIR)
++    set(SIMPLEINI_INCLUDE_DIRS "${SIMPLEINI_INCLUDE_DIR}" CACHE PATH "SimpleINI include dir" FORCE)
++    include_directories(${SIMPLEINI_INCLUDE_DIRS})
++endif()
++
++find_path(STB_INCLUDE_DIR NAMES stb_image.h PATH_SUFFIXES stb PATHS /usr/include)
++if(STB_INCLUDE_DIR)
++    include_directories(${STB_INCLUDE_DIR})
++endif()
++
++include_directories(/usr/include/SDL2)
++
++# Linker circular static archive fix
++if(UNIX AND NOT APPLE)
++    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--start-group")
++    link_libraries(-Wl,--start-group)
++    link_libraries(-Wl,--end-group)
++endif()
++
+ add_subdirectory(src/executables/sc)
+-add_subdirectory(src/executables/to)
+-add_subdirectory(src/executables/debugger)
++#add_subdirectory(src/executables/to)
++#add_subdirectory(src/executables/debugger)
+ add_subdirectory(src/executables/config)
+-if (NOT APPLE)
++if (WIN32)
+     add_subdirectory(src/executables/vrstrike)
+ endif()
+ add_subdirectory(src/executables/wc3)
